### DISCUSSION TITLE
Private dataset portuguese expecting healthier results at 10k steps

### DISCUSSION THREAD

**Discussion by JRMeyer** [coqui-ai/TTS#193] [Archived]
<USER>>>> LucasRotsen<br/> <DATE> [April 29, 2020, 5:38pm]<br/> <br/> Hi everyone!<br/> <br/> Firstly, thank you for the great TTS implementation. I was able to start<br/> training on my dataset without having any problems.<br/> <br/> Straight to the point, I'm using a private Portuguese dataset formatted<br/> like LJSpeech and my results on tensorboard are anything but healthy:<br/> <br/> A bit about config.json (only the parameters I changed from master<br/> branch):<br/> <br/> 'AÁÂÃÀBCÇDEÉÊFGHIÍJKLMNOÓÔÕPQRSTUÚVWXYZaáâãàbcçdeéêfghiíjklmnoóôpqrstuúvwxyz!'(),-.:;?'<br/> *slash* '<br/> 'ãẽĩõiyɨʉɯuɪʏʊeøɘəɵɤoɛœɜɞʌɔæɐaɶɑɒᵻʘɓǀɗǃʄǂɠǁʛpbtdʈɖcɟkɡqɢʔɴŋɲɳnɱmʙrʀⱱɾɽɸβfvθðszʃʒʂʐçʝxɣχʁħʕhɦɬɮʋɹɻjɰlɭʎʟˈˌːˑʍwɥʜʢʡɕʑɺɧɚ˞ɫ'<br/> *slash* [130000, 2, 16 *slash* ], *slash* [290000, 1, 8 *slash* ] *slash* ]<br/> <br/> I'm binge reading posts from discourse and I saw somewhere that, with<br/> gradual training, the alignments graph should start to look like a<br/> diagonal anywhere near to 10k steps. Any ideas about what could be<br/> wrong? (Or maybe I just need more training).<br/> <br/> [This is an archived TTS discussion thread from discourse.mozilla.org/t/private-dataset-portuguese-expecting-healthier-results-at-10k-steps]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> dkreutz<br/> <DATE> [April 29, 2020, 10:00pm]<br/> <br/> How many phrases/wavs are in your dataset and did you run the<br/> 'dataset_analysis' (notebooks and scripts)? *slash* <br/> You may want to convert your wav-files to a lower sampling rate, e.g.<br/> 22050 and try toggling (set to true if false and vice versa)<br/> parameters`trim_silence` and `forward_attention`<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> LucasRotsen<br/> <DATE> [April 30, 2020, 2:52am]<br/> <br/> !<br/> <br/> The dataset consists of 8286 phrases/wavs and, no, I didn't run<br/> 'dataset_analysis'. I will do that now, in conjunction with the analysis<br/> of the spectograms (results will be posted here). Could you explain why<br/> toggling both `trim_silence` and `forward_attention` true/false can<br/> help?<br/> <br/> Update (dataset_analysis execution):<br/> [AnalyzeDatasetOut.zip](https://discourse.mozilla.org/uploads/short-url/hbDDRWVxrMVxTqHb3ao0Ou6em0B.zip)<br/> (84,1,KB)<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> kms<br/> <DATE> [May 5, 2020, 3:07am]<br/> <br/> Have you advanced? If so, could you share the parameters that worked?<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> LucasRotsen<br/> <DATE> [May 5, 2020, 11:50pm]<br/> <br/> I did the changes suggested by<br/> (converted the audio to a<br/> lower sample rate, setted `do_trim_silence` to False and<br/> `forward attention` to True). I also changed `attention norm` to *slash* '<br/> softmax *slash* '.<br/> <br/> All this led to better graphs, but the model is not capable to<br/> generalize yet, and the generated audio quality is not good (audio<br/> exploding in silence).<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> dkreutz<br/> <DATE> [May 6, 2020, 9:42am]<br/> <br/> You are probably a little bit too impatient. To my experience you will<br/> see proper alignment and first meaningful audio at step 15-20k earliest.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> LucasRotsen<br/> <DATE> [May 6, 2020, 2:05pm]<br/> <br/> I've trained other voice synthesis models before, and by this time (70K<br/> steps) I'd already have something to listen to - although I know that<br/> this is not a good enough metric for comparisons. I think my impatience<br/> is because If there's something wrong with my training, I'd rather stop<br/> it 'fast' and start testing another hypothesis.<br/> <br/> In FAQ is pointed 'Check model spectrograms. Especially training outputs<br/> should converge to ground truth after 10K iterations.' and in 70k<br/> iterations I still have this:<br/> <br/> Some test audios to illustrate what I'm saying:<br/> [test_audios.zip](https://discourse.mozilla.org/uploads/short-url/5ZwSSghVJcop4eGc8H7634sC6Kw.zip)<br/> (1,6,MB)<br/> <br/> Guess for now I'll just wait some more while I try to find out what<br/> could be leading to problems (maybe is something DSP related).<br/> <br/> Thank you again! Your suggestions really helped!<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> dkreutz<br/> <DATE> [May 6, 2020, 6:58pm]<br/> <br/> I had another look at your dataset analysis. It seems like you have<br/> quite long phrase with up to 800 characters? That might cause the<br/> attention/stop problem you currently have. *slash* <br/> can have a look and<br/> comment on it? (dataset analysis is attached 3 post above).<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> erogol<br/> <DATE> [May 7, 2020, 12:10pm]<br/> <br/> what you mean model exploding at silences? You mean the random noise at<br/> the end of the spectrogram ?<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> LucasRotsen<br/> <DATE> [May 7, 2020, 1:58pm]<br/> <br/> , exactly that. I've set<br/> `do_trim_silence` to false, so I imagine it could be due to that. What<br/> really concerns me is the inability to generalize. Attention seems ok to<br/> me, so I have no idea why this is happening.<br/> <br/> Generated audio samples (the voice is just making sounds that look like<br/> words.):<br/> [test_audios.zip](https://discourse.mozilla.org/uploads/short-url/f4TXy4GJdMR9DaoAL5lKBFYgkKr.zip)<br/> (2,1,MB)<br/> <br/> The content of `test_sentences` are also converted tho phonemes before<br/> it is fed to the model, right? The 'mumbling' makes no sense to me.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> LucasRotsen<br/> <DATE> [May 8, 2020, 3:28pm]<br/> <br/> Hi guys! I just noticed that, since first steps of training, the message<br/> `Decoder stopped with max_decoder_steps` is popping when synthesizing<br/> pointed,<br/> my dataset contains big phrases (since I'll need to synthesize big<br/> sentences with model). In that case, what would be the best approach to<br/> have a successful model? I mean, would a model trained with small<br/> sentences be capable of synthesizing longer ones? I also didn't quite<br/> understand what would happen if I increased `max_decoder_steps`. Could<br/> long silences or noise be generated after the synthesized sentence?<br/> Thank you again for your time and help.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> dkreutz<br/> <DATE> [May 8, 2020, 8:27pm]<br/> <br/> Can you give examples of „big sentences'?<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> LucasRotsen<br/> <DATE> [May 8, 2020, 9:03pm]<br/> <br/> Sure! Some examples from dataset:<br/> <br/> aquele senhor estivera lá uma hora antes; ao ver que ela não estava<br/> e que não voltaria logo, deixou um pequeno pacote enviado por uma<br/> das irmãs dele e foi embora.<br/> dia consiga expressar-se tão bem contando apenas com a própria<br/> capacidade e, no entanto, este não é o estilo de uma mulher.<br/> <br/> Examples of phrases I'd like to synthesize:<br/> <br/> demonstrou convincentemente que vai participar na mudança de<br/> alternativas às soluções ortodoxas.<br/> descentralizadoras faz parte de um processo de gerenciamento do<br/> retorno esperado a longo prazo.<br/> <br/> I imagine that batch synthesis could be a solution to this, but I fear<br/> that the delay in the synthesis time would be too long.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> dkreutz<br/> <DATE> [May 8, 2020, 9:37pm]<br/> <br/> According to [this](https://github.com/NVIDIA/tacotron2/issues/184) you<br/> try to increase max_decoder_steps. *slash* <br/> But again - the dataset analysis show that you have phrases much longer<br/> than 200 characters. Try to keep the maximun below 200 during initial<br/> training. You can fine tune your models with longer phrases when it has<br/> aligned...<br/> <br/> [Archived Post]<br/>
